src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -200,10 +208,59 @@ internal int SerializeLogRecord(LogRecord logRecord)
             var name = logRecord.CategoryName;
 
             // If user configured explicit TableName, use it.
-            if (this.m_tableMappings == null || !this.m_tableMappings.TryGetValue(name, out var eventName))
+            if (this.m_tableMappings == null || (!this.m_tableMappings.TryGetValue(name, out var eventName) && !this.shouldPassThruTableMappings))
             {
                 eventName = this.m_defaultEventName;
             }
+            else if (this.shouldPassThruTableMappings && eventName == null)
```

Comment:
Should we add the `eventName` generated here to `this.m_tableMappings` to avoid doing the computation over and over for the same logger?

If we do that, we could have unbounded growth for `this.m_tableMappings` if the user keeps creating new loggers.

Reply:
Right, if users have tons of categories, it can lead to unbounded growth of `this.m_tableMappings.` While, I'm not sure whether this would be a common case.
Maybe we can have a upper limit for the number of elements stored in the map, and start to do runtime calculation when the map is up to the limit?

Reply:
I've added a `ConcurrentDictionary` for caching, but I am not sure how to determine a proper upper limit for the size of the Dictionary yet...

Reply:
What's the perf benefits/downsides with vs. without caching?

Reply:
E.g. here is what I can imagine:

caching:
* Pros: might save some CPU cycles
* Cons: contention on the concurrent dictionary, extra memory usage

non-caching:
* Pros: no contention, constant memory usage (which can be zero-heap-alloc)
* Cons: might burn extra CPU cycles

What's difference if we benchmark two approaches using a typical profile/load?

Reply:
https://github.com/Yun-Ting/opentelemetry-dotnet-contrib/pull/1